bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/type/ThingTypeBuilderTest.java

```diff
@@ -0,0 +1,230 @@
+
+    @Test
+    public void withLabelAndThingTypeIdAndBindingID_shouldSetListed() {
+        ThingType thingType = builder.build();
+
+        assertThat(thingType.isListed(), is(true));
+    }
+
+    @Test
+    public void whithDescription_shouldSetDescription() {
+        ThingType thingType = builder.withDescription(DESCRIPTION).build();
+
+        assertThat(thingType.getDescription(), is(DESCRIPTION));
+    }
+
+    @Test
+    public void whithCategory_shouldSetCategory() {
+        ThingType thingType = builder.withCategory(CATEGORY).build();
+
+        assertThat(thingType.getCategory(), is(CATEGORY));
+    }
+
+    @Test
+    public void whithListed_shouldBeListed() {
+        ThingType thingType = builder.isListed(false).build();
+
+        assertThat(thingType.isListed(), is(false));
+    }
+
+    @Test
+    public void whithRepresentationProperty_shouldSetRepresentationProperty() {
+        ThingType thingType = builder.withRepresentationProperty(REPRESENTATION_PROPERTY).build();
+
+        assertThat(thingType.getRepresentationProperty(), is(REPRESENTATION_PROPERTY));
+    }
+
+    @Test
+    public void whithChannelDefinitions_shouldSetUnmodifiableChannelDefinitions() {
+        ThingType thingType = builder.withChannelDefinitions(mockList(ChannelDefinition.class, 2)).build();
+
+        assertThat(thingType.getChannelDefinitions(), is(hasSize(2)));
+        try {
+            thingType.getChannelDefinitions().add(mock(ChannelDefinition.class));
+            fail();
+        } catch (UnsupportedOperationException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void whithChannelGroupDefinitions_shouldSetUnmodifiableChannelGroupDefinitions() {
+        ThingType thingType = builder.withChannelGroupDefinitions(mockList(ChannelGroupDefinition.class, 2)).build();
+
+        assertThat(thingType.getChannelGroupDefinitions(), is(hasSize(2)));
+        try {
+            thingType.getChannelGroupDefinitions().add(mock(ChannelGroupDefinition.class));
+            fail();
+        } catch (UnsupportedOperationException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void withProperties_shouldSetUnmodifiableProperties() {
+        ThingType thingType = builder.withProperties(mockProperties()).build();
+
+        assertThat(thingType.getProperties().entrySet(), is(hasSize(2)));
+        try {
+            thingType.getProperties().put("should", "fail");
+            fail();
+        } catch (UnsupportedOperationException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void whithConfigDescriptionURI_shouldSetConfigDescriptionURI() throws Exception {
+        ThingType thingType = builder.withConfigDescriptionURI(new URI(CONF_URI)).build();
+
+        assertThat(thingType.getConfigDescriptionURI(), is(new URI(CONF_URI)));
+    }
+
+    @Test
+    public void withExtensibleChannelTypeIds_shouldSetUnmodifiableExtensibleChannelTypeIds() {
+        @SuppressWarnings("null")
```

Comment:
Is this really needed with the global suppression on class-level?

Reply:
no, obsolete.